### PR TITLE
[NeoMathEngine] pooling layers formatting, whitespaces, const modifiers

### PR DESCRIPTION
**Pull request by favorart**
* For *PoolngBackward functions rename arguments as SourceDiff and ResultDiff
* Updated the files license strings to 2023
* Appended 'const' modifiers, where necessary
* Whitespaces formatting fix

### CODE REVIEW

File: NeoMathEngine/include/NeoMathEngine/NeoMathEngine.h [hunk 1]
```diff
@@ -835,16 +835,16 @@ class NEOMATHENGINE_API IDnnEngine : public IBlasEngine {
...
 
 	// GlobalMaxOverTime-Pooling
 	// The descriptor should be destroyed using the standard delete operator after use.
 	virtual CGlobalMaxOverTimePoolingDesc* InitGlobalMaxOverTimePooling( const CBlobDesc& source, const CBlobDesc& result ) = 0;
 
 	virtual void BlobGlobalMaxOverTimePooling( const CGlobalMaxOverTimePoolingDesc& desc, const CFloatHandle& source, const CIntHandle* maxIndices,
 		const CFloatHandle& result ) = 0;
-	virtual void BlobGlobalMaxOverTimePoolingBackward( const CGlobalMaxOverTimePoolingDesc& desc, const CFloatHandle& source, const CIntHandle& maxIndices,
-		const CFloatHandle& result ) = 0;
+	virtual void BlobGlobalMaxOverTimePoolingBackward( const CGlobalMaxOverTimePoolingDesc& desc, const CFloatHandle& sourceDiff, const CIntHandle& maxIndices,
```
Comment (FedyuninV) [inline NeoMathEngine/include/NeoMathEngine/NeoMathEngine.h]: Wrong names! See `CMaxOverTimePoolingLayer::BackwardOnce`. Also fix all the `IMathEngine` implementations you've changed

Comment (favorart) [reply NeoMathEngine/include/NeoMathEngine/NeoMathEngine.h]: Do I understand correct, that in the original signature the arguments' names were reversed
```
MathEngine().BlobMaxOverTimePoolingBackward( *desc, outputDiffBlobs[0]->GetData(),
			maxIndices->GetData<int>(), inputDiffBlobs[0]->GetData() );
```
So should be correct
```
void BlobGlobalMaxOverTimePoolingBackward( const CGlobalMaxOverTimePoolingDesc& desc,
			const CFloatHandle& resultDiff, const CIntHandle& maxIndices, const CFloatHandle& sourceDiff ) = 0;
```

Comment (FedyuninV) [reply NeoMathEngine/include/NeoMathEngine/NeoMathEngine.h]: Yep. Yet another backward function calls sourceDiff based on resultDiff.

Comment (favorart) [reply NeoMathEngine/include/NeoMathEngine/NeoMathEngine.h]: To be sure I marked all `source` data in forward pooling `const`, also `resultDiff` and `maxIndex` data in backward pooling `const` too. So it becomes
```
void Blob*PoolingBackward( const C*PoolingDesc& desc,
	const CConstFloatHandle& resultDiff, const CConstIntHandle& maxIndices, const CFloatHandle& sourceDiff );
```

**Review by FedyuninV — APPROVED** [review]

**Pull request merged**